python/pyspark/taskcontext.py

```diff
@@ -147,8 +147,8 @@ def __init__(self):
     @classmethod
     def _getOrCreate(cls):
         """Internal function to get or create global BarrierTaskContext."""
-        if cls._taskContext is None:
-            cls._taskContext = BarrierTaskContext()
+        if not isinstance(cls._taskContext, BarrierTaskContext):
+            cls._taskContext = object.__new__(cls)
```

Comment:
BTW, I think we should just `BarrierTaskContext()`. Let's don't make it complicated next time.

Reply:
ah good point! @xuanyuanking can you send a small followup?

Reply:
I think the answer is no, maybe I was not clear enough in my previous explain  https://github.com/apache/spark/pull/22962#discussion_r232528333, use `BarrierTaskContext()` here is my first commit https://github.com/apache/spark/pull/22962/commits/0cb2cf6e9ece66861073c31b579b595a9de5ce81 , it should also need to rewrite `__new__` for `BarrierTaskContext`, otherwise the bug still exists cause its parent class `TaskContext` rewrite `__new__()`, when we call `BarrierTaskContext()` here in a reused worker, a `TaskContext` instance will be returned in `TaskContext.__new__()`:https://github.com/apache/spark/blob/c00e72f3d7530eb2ae43d4d45e8efde783daf6ff/python/pyspark/taskcontext.py#L47

Reply:
could you add some comments to explain it? so that people won't get confused again.

Reply:
Can we get rid of the rewrite all? It's never a good idea to overwrite them unless it's absolutely required.

Reply:
Also, we should remove __init__ too. That's what Python interpretor will implicitly insert for both.

Reply:
Also, next time please fully describe what's going on in PR description. Even I was confused about it and misread that `__new__` is actually being inherited.

Reply:
```
could you add some comments to explain it?
```
@cloud-fan Sorry for the less explain and more comments should be done at first, will done in follow up PR.
```
Can we get rid of the rewrite all?
we should remove __init__ too
next time please fully describe what's going on in PR description
```
@HyukjinKwon Sorry for the less explain, all these will be done in next follow up PR, and the new UT.

Reply:
```
could you add some comments to explain it?
```
@cloud-fan Sorry for the less explain and more comments should be done at first, will done in follow up PR.
```
Can we get rid of the rewrite all?
we should remove __init__ too
next time please fully describe what's going on in PR description
```
@HyukjinKwon Sorry for the less explain, all these will be done in next follow up PR, and the new UT.